examples/llm/src/mosaic_gpt.py

```diff
@@ -421,10 +421,12 @@ def forward(self,
                                     seq_len=S,
                                     key_padding_mask=key_padding_mask,
                                     dtype=x.dtype)
+        if self.cfg.attn_impl == 'flash':
+            key_padding_mask = torch.ones_like(input_ids, dtype=torch.bool)
+        if self.cfg.attn_impl == 'triton':
+            key_padding_mask = None
         for block in self.transformer.blocks:  # type: ignore
-            x = block(
-                x, None if self.cfg.attn_impl == 'triton' else key_padding_mask,
-                attn_mask)
+            x = block(x, key_padding_mask, attn_mask)
```

Comment:
Should the triton path still receive None? Also by moving to the pre-tokenized datasets, we no longer support tokenization on the fly with MosaicGPT models right? as for tokenization on the fly setting key_padding_mask to 1 is not right.

Reply:
The `triton` path with `None` is just copying what was there before, I am not sure of its correctness. @vchiley thoughts here?

For on the fly padding and tokenization you are right, that is a bug on my part. Will modify the check appropriately.

Reply:
triton doesn't support key_padding_mask, if key_padding_mask shows invalid tokens, you need to handle it as part of the attn_bias (at least it wasn't supported last time i looked at it; with time that might change; i there is an [issue in HazyR](https://github.com/HazyResearch/flash-attention/issues/104) asking for it to be supported)